src/Microsoft.ML.Core/Data/MetadataUtils.cs

```diff
@@ -447,21 +450,22 @@ public static bool TryGetCategoricalFeatureIndices(Schema schema, int colIndex,
                 {
                     int previousEndIndex = -1;
                     isValid = true;
-                    for (int i = 0; i < catIndices.Values.Length; i += 2)
```

Comment:
>catIndices.Values.Length [](start = 40, length = 24)

Oh that's nice. This was a bug. :p

Reply:
Is it possible that the difference we saw in `FastTreeBinaryClassificationCategoricalSplitTest` was due to you fixing this bug? Or are you quite certain that the difference is due to the code inside the transposer?

---
In reply to: [232425411](https://github.com/dotnet/machinelearning/pull/1580#discussion_r232425411) [](ancestors = 232425411)